client/asset/btc/spv.go

```diff
@@ -1039,17 +1138,27 @@ func (w *spvWallet) startWallet() error {
 		bailOnWallet()
 	}
 
-	// If we're on regtest and the peers haven't been explicitly set, add the
-	// simnet harness alpha node as an additional peer so we don't have to type
-	// it in.
-	if w.chainParams.Name == "regtest" && len(w.connectPeers) == 0 {
-		w.connectPeers = append(w.connectPeers, "localhost:20575")
+	// Depending on the network, we add some addpeers or a connect peer. On
+	// regtest, if the peers haven't been explicitly set, add the simnet harness
+	// alpha node as an additional peer so we don't have to type it in. On
+	// mainet and testnet3, add a known reliable persistent peer to be used in
+	// addition to normal DNS seed-based peer discovery.
+	var addPeers []string
+	switch w.chainParams.Net {
+	case wire.MainNet:
+		addPeers = []string{"cfilters.ssgen.io"}
```

Comment:
I wonder whether this could result in bottlenecks. What are the chances that this node is picked for the filter source? Is it random, or since this is in addpeer, does it result in this node being chosen more often or even always? Hard-coding a mainnet node also feels pretty icky, though I admit I've considered it as a solution for various problems. I would hope that if we do have "sanctioned" nodes, we would make them opt-out at the very least.

Reply:
Good questions.  The "addpeer" nodes are only suggestions, so they get mixed into the other nodes from DNS seeders.  I had the same icky feeling at first, but when you consider we are told what nodes to use in part from the Bitcoin Core devs, some of whom were opposed to compact filters, it's quite fair that we at least throw one in the mix.

https://github.com/btcsuite/btcd/blob/7b6c2b342379f3cba880d9f3cc14c77fc7500969/chaincfg/params.go#L261-L266

I'm not sure opting out provides much benefit considering that we cannot opt out of the DNS seed nodes either.  This could be prevented however if we expose the `connect` peers, which are exclusive and prevent discovery.

Regarding if the addpeer takes some priority or gets chosen more readily, I cannot tell for sure, but in my observations it does seem to take some precedence.